storage/scratchpad/src/sparse_merkle/mod.rs

```diff
@@ -509,6 +509,393 @@ where
         ret
     }
 
+    /// The batch version of `serial_update`. The function will return a Sparse Merkle Tree of the final state.
+    /// This method is optmized to avoid repetitive common path updating and parallelize the processing as much
+    /// as possible.
+    pub fn batch_update(
```

Comment:
This is actually the batch version of `update()` :)  
After this diff `update()` can be removed, not the old `batch_update()`